### DISCUSSION TITLE
Issues with MCP and Windows Desktop App

### DISCUSSION THREAD

**Discussion by PederHP** [modelcontextprotocol/.github#17] [Q&A] [answered]
I am experiencing two issues with MCP support via the Windows Application:

### Issue 1: MCP Connection Issue
**Current Behavior:**
1. NPX commands successfully launch console windows running MCP servers
2. Error messages "Could not attach to MCP server memory/filesystem" appear
3. Missing MCP Connect icon in UI

**Troubleshooting Steps Attempted:**
1. Running application with elevated permissions
2. Direct launch vs Squirrel bootstrap
3. Running in Developer mode

**Logs:**
```
2024-11-26T15:34:55.458Z [info] Attempting to connect to MCP server memory...
2024-11-26T15:34:55.460Z [info] Attempting to connect to MCP server filesystem...
2024-11-26T15:34:55.505Z [info] Connected to MCP server memory!
2024-11-26T15:34:55.510Z [info] Connected to MCP server filesystem!
```

### Issue 2: Settings Not Loaded
**Steps to Reproduce:**
1. Edit claude_desktop_config.json with MCP server configuration
2. Try to restart application normally
3. Observe that changes aren't reflected
4. Must use Task Manager to kill process for changes to take effect

**Expected Behavior:** Configuration changes should be detected and loaded on normal application restart

**Environment Details:**
Windows 10
Claude Desktop App version: 0.7.1
Configuration file location: C:\Users\<me>\AppData\Roaming\Claude

**Comment by PederHP** [comment, marked as answer]
Found the work around about using a global node install and then `node` instead of `npx -y`. It did the trick.

**Reply by PederHP** [reply]
Remember to use the full path for node (to the index.ts) rather than the same relative one that npx takes.

**Comment by deepseven** [comment]
Do you mean running
> npm install -g @modelcontextprotocol/server-filesystem

from the command line and then using 
```
{
	"mcpServers": {
		"filesystem": {
		  "command": "node",
		  "args": [ 
			"@modelcontextprotocol/server-filesystem", 
			"<path to folder>"
			]
		}
}
```
in claude_desktop_config.json? 

It didn't work for me but maybe I'm doing something wrong. (node.js newbie here 🙋)

**Reply by juanma1331** [reply]
Hi, I'm having the same issues too.

**Reply by PederHP** [reply]
You need to use the full path to the index.ts file (instead of the @...) The global node install should place it somewhere reachable. Claude can help you with the correct path on your system.

**Reply by deepseven** [reply]
Thank you @PederHP ! I followed your advice and the steps mentioned in the link below, and it worked: https://github.com/modelcontextprotocol/servers/issues/40#issuecomment-2504754720

**Reply by brandtcormorant** [reply]
instead of using the `node` command you can instead use the `npx` command to run the server:

```json
{
  "mcpServers": {
    "filesystem": {
      "command": "npx",
      "args": [
        "@modelcontextprotocol/server-filesystem",
        "<path to folder>"
      ]
    }
  }
}
```

it's just necessary to install the package globally first:

```
npm install -g @modelcontextprotocol/server-filesystem
```

**Comment by jspahrsummers** [comment]
Sorry, this is the same issue as https://github.com/modelcontextprotocol/servers/issues/40. We're working on a fix. 🙏

**Reply by Jacck** [reply]
any progress?
